src/qibolab/result.py

```diff
@@ -1,53 +1,64 @@
 from dataclasses import dataclass
 from functools import cached_property
+from typing import Optional
 
 import numpy as np
 import numpy.typing as npt
-
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64), ("shots", np.uint32)])
+from scipy import signal
 
 
 @dataclass
 class ExecutionResults:
     """Data structure to deal with the output of execute_pulse_sequence"""
 
-    array: npt.NDArray[ExecRes]
+    _i: npt.NDArray[np.float64]
+    _q: npt.NDArray[np.float64]
```

Comment:
After discussing with @AleCandido, we could still keep the `ExecRes` for i and q and have arbitrary dimensions. In fact the issue was in the `from_components` constructor (see comment below). That being said, I am not sure if `ExecRes` is preferrable to the current solution. Either way, for sure `shots` should be a seperate attribute.

Reply:
Ok, so if I understand correctly I will restore the `ExecRes` array which will contain only i and q and I will keep `shots` as a separate argument (as a standard `np.ndarray`).